### DISCUSSION TITLE
Use Postgres connection pooling for Nominatim

### DISCUSSION THREAD

**Discussion by shumailxyz** [mediagis/nominatim-docker#570] [Q&A]
I did a fresh install of Nominatim 4.4 along with [Postgis 15-3.4](https://registry.hub.docker.com/r/postgis/postgis). Most of the things work fine as expected however one of my applications broke. 

That application basically was doing around 500 reverse geocoding requests on `/reverse.php` concurrently. It was working fine with Nominatim 4.2 and Postgres 11 with Postgis 2.5 (was using the docker image [mdillon/postgis](https://hub.docker.com/r/mdillon/postgis)).

With the updated installation, around 100 (probably due to the default value of max_connections = 100 in postgresql.conf) out of 500 requests are successful and for the rest, nominatim returnes `500 internal server error`. I didn't find anything in Nominatim or apache2 logs but when Ichecked the Postgres logs, i saw these logs:

```
FATAL:  sorry, too many clients already
FATAL:  sorry, too many clients already
```

Did something change in newer version in regards to how connection with Postgres is initialized? Should it be using single client to initiate the connection pool with Postgres and reuse connection? Would appreciate if someone has any pointers on how to approach this. Thanks!

**Comment by mtmail** [comment]
Duplicate of https://github.com/osm-search/Nominatim/discussions/3477

**Reply by shumailxyz** [reply]
I think this can be closed in favor of the other one. Unfortunately, Github doesn't provide a way to delete/close a discussion thread but may be the maintainers can close this.
